cdap-app-fabric/src/test/java/co/cask/cdap/internal/app/services/http/handlers/WorkflowHttpHandlerTest.java

```diff
@@ -401,54 +416,116 @@ public void testWorkflowForkApp() throws Exception {
     Id.Program programId = Id.Program.from(TEST_NAMESPACE2, workflowAppWithFork, ProgramType.WORKFLOW,
                                            workflowWithFork);
 
-    Map<String, String> runtimeArguments = ImmutableMap.of("done.file", doneFile.getAbsolutePath(),
-                                                           "oneaction.file", oneActionFile.getAbsolutePath(),
-                                                           "anotheraction.file", anotherActionFile.getAbsolutePath());
-
-    setAndTestRuntimeArgs(programId, runtimeArguments);
+    setAndTestRuntimeArgs(programId, runtimeArgs);
 
+    // Start a Workflow
     startProgram(programId, 200);
 
+    // Workflow should be running
     waitState(programId, "RUNNING");
 
+    // Get the currently running RunRecord for the Workflow
     List<RunRecord> historyRuns = getProgramRuns(programId, "running");
     Assert.assertTrue(historyRuns.size() == 1);
+    RunRecord record = historyRuns.get(0);
+    String runId = record.getPid();
 
-    String runId = historyRuns.get(0).getPid();
+    // Wait till first action in the Workflow starts executing
+    while (!firstSimpleActionFile.exists()) {
```

Comment:
Incase of failure we will never come out of the loop. There should be a max retry and throw exception if the file doesn't exist after the retry